src/main/java/teammates/common/util/FieldValidator.java

```diff
@@ -103,149 +73,107 @@
+            "\"${userInput}\" is not acceptable to TEAMMATES as ${fieldName} because it ${reason}.";
+    public static final String HINT_FOR_CORRECT_FORMAT_FOR_SIZE_CAPPED_NON_EMPTY =
+            "The value of ${fieldName} should be no longer than ${maxLength} characters. It should not be empty.";
+    public static final String HINT_FOR_CORRECT_FORMAT_FOR_SIZE_CAPPED_POSSIBLY_EMPTY =
+            "The value of ${fieldName} should be no longer than ${maxLength} characters.";
+    public static final String HINT_FOR_CORRECT_FORMAT_FOR_INVALID_NAME =
+            "All ${fieldName} must start with an alphanumeric character, and cannot contain any vertical bar "
+            + "(|) or percent sign (%).";
+
+    // generic (i.e., not specific to any field) error messages
+    public static final String SIZE_CAPPED_NON_EMPTY_STRING_ERROR_MESSAGE =
+            ERROR_INFO + " " + HINT_FOR_CORRECT_FORMAT_FOR_SIZE_CAPPED_NON_EMPTY;
+    public static final String SIZE_CAPPED_POSSIBLY_EMPTY_STRING_ERROR_MESSAGE =
+            ERROR_INFO + " " + HINT_FOR_CORRECT_FORMAT_FOR_SIZE_CAPPED_POSSIBLY_EMPTY;
+    public static final String INVALID_NAME_ERROR_MESSAGE =
+            ERROR_INFO + " " + HINT_FOR_CORRECT_FORMAT_FOR_INVALID_NAME;
+    public static final String WHITESPACE_ONLY_OR_EXTRA_WHITESPACE_ERROR_MESSAGE =
+            "The provided ${fieldName} is not acceptable to TEAMMATES as it contains only whitespace "
+            + "or contains extra spaces at the beginning or at the end of the text.";
+    public static final String NON_HTML_FIELD_ERROR_MESSAGE =
+            Sanitizer.sanitizeForHtml("The provided ${fieldName} is not acceptable to TEAMMATES as it cannot contain"
+                                      + " the following special html characters in brackets: (< > \\ / ' &)");
+    public static final String NON_NULL_FIELD_ERROR_MESSAGE =
+            "The provided ${fieldName} is not acceptable to TEAMMATES as it cannot be empty.";
+
+    // field-specific error messages
+    public static final String HINT_FOR_CORRECT_EMAIL =
+            "An email address contains some text followed by one '@' sign followed by some more text. "
+            + "It cannot be longer than ${maxLength} characters. "
+            + "It cannot be empty and it cannot have spaces.";
+    public static final String EMAIL_ERROR_MESSAGE =
+            ERROR_INFO + " " + HINT_FOR_CORRECT_EMAIL;
+    public static final String EMAIL_TAKEN_MESSAGE =
+            "Trying to update to an email that is already used by: %s/%s";
+
+    public static final String HINT_FOR_CORRECT_COURSE_ID =
+            "A Course ID can contain letters, numbers, fullstops, hyphens, underscores, and dollar signs. "
+            + "It cannot be longer than ${maxLength} characters. "
+            + "It cannot be empty or contain spaces.";
+    public static final String COURSE_ID_ERROR_MESSAGE =
+            ERROR_INFO + " " + HINT_FOR_CORRECT_COURSE_ID;
```

Comment:
Noted. But as per above, I'm trying not to make any changes in wording in this PR.